test/cql-pytest/test_materialized_view.py

```diff
@@ -288,6 +288,30 @@ def test_is_not_null_forbidden_in_filter(cql, test_keyspace, cassandra_bug):
         finally:
             cql.execute(f"DROP MATERIALIZED VIEW IF EXISTS {test_keyspace}.{mv}")
 
+# In Cassandra's implementation of materialized views, base replicas use a batchlog
+# to increase durability of view updates. Similarly, Scylla uses hinted handoff
+# to replay failed view updates. In both cases, batchlog entries / hints expire
+# after gc_grace_seconds of the view. Therefore, it doesn't make sense to set
+# gc_grace_seconds to a small value as this causes entries / hints to expire
+# immediately.
```

Comment:
I found this text hard to believe, so I looked, and, you're right, Cassandra does exactly this crazy thing. I even found a document on it - https://thelastpickle.com/blog/2018/03/21/hinted-handoff-gc-grace-demystified.html - but I still don't understand WHY they do it.

But now I don't understand - this crazy behavior breaks not just gc_grace_seconds but any "small" gc_grace_seconds - even one hour is too little if there is a chance that hints need to be replayed after one hour.

Also, when hints are enabled on a base table - not a view, we should have the same problem! Why is this check only for views?

It sounds to me like we should either:
1. Rethink this weird gc_grace_seconds-used-to-expire-hints thing.
or
2. Add a "guardrails"-like configuration parameter of "minimum_gc_grace_seconds", with lower numbers causing an error. We could set the default minimum to, say, one day, but allow us to use 0 as the guardrail setting in tests to make these tests fast.

Reply:
As I said [above](https://github.com/scylladb/scylladb/pull/20618#discussion_r1774355613), the tombstone gc grace period, is based on having repair run in the grace period to ensure data and tombstones are synchronized, making tombstone garbage collection safe.
And repair should flush hints (and batchlog).
Sadly, repair does not run when rf=1, making immediate tombstone garbage collection somewhat sensible in this configuration.  But this might expose a hole in the design if hints or batchlog may replay deleted data after tombstones were gc'ed.